test/misc.jl

```diff
@@ -1234,4 +1234,8 @@ end
 
 @testset "Base/timing.jl" begin
     @test Base.jit_total_bytes() >= 0
+
+    # sanity check `@allocs` returns what we expect in some very simple cases
+    @test (@alloc "a") == 0
+    @test (@alloc "a" * "b") == 1
```

Comment:
i didn't see any tests for `@allocated` so wasn't sure how to approach testing... advice appreciated!